### DISCUSSION TITLE
How would I make it so nextcloud uses my second hard disk for data

### DISCUSSION THREAD

**Discussion by ixanza** [nextcloud/all-in-one#597] [Questions] [answered]
Output of df-h

```
Filesystem                         Size  Used Avail Use% Mounted on
tmpfs                              1.6G  2.4M  1.6G   1% /run
/dev/mapper/ubuntu--vg-ubuntu--lv   98G   15G   79G  16% /
tmpfs                              7.8G     0  7.8G   0% /dev/shm
tmpfs                              5.0M     0  5.0M   0% /run/lock
/dev/nvme0n1p2                     2.0G  144M  1.7G   8% /boot
/dev/nvme0n1p1                     1.1G  5.3M  1.1G   1% /boot/efi
/dev/sda1                          916G  1.4G  868G   1% /home
tmpfs                              1.6G  4.0K  1.6G   1% /run/user/1000
```

My docker-compose file

```version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
  caddy_data:
    external: true
  caddy_config:

services:
  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 8080:8080
    environment:
      - APACHE_PORT=11000
      - NEXTCLOUD_DATADIR=/mnt/home/user/ncdata
      - NEXTCLOUD_MOUNT=/mnt/home/user
    depends_on:
      - caddy

  caddy:
    image: caddy:alpine
    restart: unless-stopped
    container_name: caddy
    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile
      - ./certs:/certs
      - ./caddy_config:/config
      - ./caddy_data:/data
      - ./sites:/srv
    network_mode: "host"
```

However when looking at the free space in my nextcloud it seems to be using mnt/ncdata which only has 98GB 
![image](https://user-images.githubusercontent.com/5535758/167279416-e31587e9-ff1d-4af8-8b78-b2f9a71d7851.png)

How would I make i so ncdata uses my 1TB drive?

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir

**Reply by ixanza** [reply]
It's set as /mnt/home/user/ncdata  but its using /mnt/ncdata still

under environment:
      - NEXTCLOUD_DATADIR=/mnt/home/user/ncdata
      - NEXTCLOUD_MOUNT=/mnt/home/user

tbh I think it might be my LVM configuration messing it up. It's a fairly fresh setup of ubuntu server I might set it up on windows again where I'm more comfortable with the filesystem.

**Reply by szaimen** [reply]
> It's set as /mnt/home/user/ncdata but its using /mnt/ncdata still

Are you sure?
What is the output of
`sudo ls -l /mnt/home/user/ncdata` ?

**Reply by ixanza** [reply]
huh it is using that i guess..

```
total 16
drwxr-xr-x  4 www-data www-data 4096 May  7 16:47 <admin user>
drwxr-xr-x 10 www-data www-data 4096 May  7 16:55 appdata_ocnwuteozurw
-rw-r--r--  1 www-data www-data    0 May  7 16:44 audit.log
drwxr-xr-x  6 www-data www-data 4096 May  7 17:27 <regular user>
drwxr-xr-x  3 www-data www-data 4096 May  7 16:57 files_encryption
-rw-r--r--  1 www-data www-data    0 May  7 16:44 index.html
-rw-r--r--  1 www-data www-data    0 May  7 16:44 nextcloud.log
```

But the free space is weird. it only has 93GB space but mnt/home/user is on a 1TB hdd. how would I get it to use all of that?

**Reply by szaimen** [reply]
It looks like you chose the wrong path.

You should have used `NEXTCLOUD_MOUNT=/home/user`
Instead of `NEXTCLOUD_MOUNT=/mnt/home/user`
Now all the data files are stored on the root drive.

**Reply by szaimen** [reply]
We'll allow this path in the future. See https://github.com/nextcloud/all-in-one/pull/584

**Reply by ixanza** [reply]
Alrighty thank you

**Reply by szaimen** [reply]
Sorry just saw that obviously `NEXTCLOUD_DATADIR=/home/user/ncdata` needs to be used as well

**Reply by ixanza** [reply]
Thank you. I don't have any files stored yet so I will try incorporating this after work.

**Reply by S0ulf3re** [reply]
Hey, I don't know much about docker. How do you change the variables of a master container? I keep trying to switch into the master container and running the env command but that doesn't keep it.

I should note that I am using the 1-click install for Nextcloud on Digitalocean

**Reply by szaimen** [reply]
Hello, please open a new thread for this. Thanks!
See https://github.com/nextcloud/all-in-one/discussions/new?category=questions
